### PR TITLE
Memory Write implementation for AXAPI v21/v30

### PR DESCRIPTION
**Pull request by mdurrant-b3**
Functional memory write implementation for both AXAPI versions.  Ensures active and alternate shared partitions are written.

### CODE REVIEW

**Comment by mdurrant-b3** [comment]
check

**Comment by mdurrant-b3** [comment]
Requires https://github.com/a10networks/acos-client/pull/109 to be merged before it will pass.

**Comment by mdurrant-b3** [comment]
check

**Comment by mdurrant-b3** [comment]
Failure appears to be related to pulling outdated acos-client, restarting.

**Comment by mdurrant-b3** [comment]
I was wrong.  There was a bug in acos-client.  acos-client PR https://github.com/a10networks/acos-client/pull/115 does a terrible job of dealing with this issue.

The problem is when the A10DeleteContext exists, it runs the "if we have no more LB elements, delete the partition" routine.  So when we go to activate the partition that we have stored as current partition_name`), it fails but the memory write command succeeds.

**Comment by mdurrant-b3** [comment]
Local tests indicate this will pass CI once https://github.com/a10networks/acos-client/pull/115 is merged.

**Comment by mdurrant-b3** [comment]
check

**Comment by mdurrant-b3** [comment]
Failed due to not having latest acos-client in PyPi (my fault for not making sure that was the case before kicking off the job).

**Comment by mdurrant-b3** [comment]
check

File: a10_neutron_lbaas/a10_context.py [hunk 1]
```diff
@@ -86,7 +88,10 @@ class A10WriteContext(A10Context):
     def __exit__(self, exc_type, exc_value, traceback):
         if exc_type is None and self.device_cfg.get('write_memory', True):
             try:
-                self.client.system.action.write_memory()
+                partition_deleted = getattr(self, "partition_deleted", False)
```
Comment (dougwig) [inline a10_neutron_lbaas/a10_context.py]: I'd suggest setting this to False in __init__, so this weird construct doesn't have to exist.  :-)

**Comment by dougwig** [comment]
I'm gonna merge this, but I won't promise not to remove that nasty getattr() later.

**Pull request merged**